pandas/tests/plotting/test_xlim_ylim_dtype.py

```diff
@@ -0,0 +1,51 @@
+import pytest
+import pandas as pd
+from datetime import datetime as dt
+import numpy as np
+
+"""targeting #GH40781"""
+
+# arbitrary df to test
+df = pd.DataFrame(
+    {
+        "A" : [1, 2, 3, 4, 5, 6, 7, 8, 9, 10],
+        "B": [dt.now() for i in range(0, 10)]
+    }
+)
+
+
+def test_axis_lim_invalid():
+    global df
+    """
+    supplying ranges with str, datetime and
+    mixed int,str and float,str dtype to raise
+    ValueError. Valid dtypes are np.datetime64
+    int, and float.
+    """
+    lim = [
+        ["1", "3"],
+        [dt.now(), dt.now()],
+        [1, "2"],
+        [0.1, "0.2"],
+        [np.datetime64(dt.now()), dt.now()]
+    ]
```

Comment:
could you try to parametrize?

Reply:
The tests are parametrized now

Reply:
I guess that @charlesdong1991 was referring to use of ``pytest.mark.parametrize``.
Or did you forget to push new changes?

Reply:
Thank you @ivanovmg, I have used following parametrization to create a custom dataframe, and I can see I have comitted it in latest commit. Can you please confirm it?
```
@pytest.mark.parametrize("df, lim", [
    (
        DataFrame(
            {
                "A" : [1, 2, 3, 4, 5, 6, 7, 8, 9, 10],
                "B": [dt.now() for i in range(0, 10)]
            }
        ),
        [
            ["1", "3"],
            [dt.now(), dt.now()],
            [1, "2"],
            [0.1, "0.2"],
            [np.datetime64(dt.now()), dt.now()]
        ]
    )
]
)
```

Reply:
Please see: https://github.com/pandas-dev/pandas/pull/40889/commits/c00977a3cff416a83c9f88abbedd4ddaaecce9c7

Reply:
Yes, I confirm. It was a mistake on my end - I looked at an outdated version.